homeassistant/components/blink/config_flow.py

```diff
@@ -18,11 +18,58 @@
 )
 from homeassistant.core import callback
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaFlowFormStep,
+    SchemaOptionsFlowHandler,
+)
 
 from .const import DEFAULT_SCAN_INTERVAL, DEVICE_ID, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
+SIMPLE_OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): int,
+    }
+)
+
+
+def _get_blink_from_handler(handler: SchemaOptionsFlowHandler) -> Blink:
+    """Extract blink instance from hass.data."""
+    entry_id = handler.config_entry.entry_id
+    blink: Blink = handler.hass.data[DOMAIN][entry_id]
+    return blink
```

Comment:
The previous implementation cached this in `async_step_init` and then used the cached value in `async_step_simple_options`

Should we replicate that using for example
```python
def SchemaCommonFlowHandler():
    def __init__():
        local_context: dict[str, Any] = {}

def _get_options_init_schema(handler: SchemaOptionsFlowHandler) -> None:
    entry_id = handler.config_entry.entry_id
    handler.local_context["blink"] = handler.hass.data[DOMAIN][entry_id]

def _validate_simple_options(
    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
) -> dict[str, Any]:
    blink: Blink = handler.local_context["blink"]
    blink.refresh_rate = user_input[CONF_SCAN_INTERVAL]
```

cc @emontnemery

Reply:
I don't think we need to complicate the `SchemaOptionsFlowHandler` handler, a config entry update listener should adjust the refresh rate instead.

Reply:
Moved to standard listener